Python/parking_lot.c

```diff
@@ -67,10 +82,17 @@ _PySemaphore_Init(_PySemaphore *sema)
         Py_FatalError("parking_lot: sem_init failed");
     }
 #else
+    #if defined(CONDATTR_MONOTONIC)
+        if (!condattr_ptr){
+            pthread_condattr_init(&condattr);
```

Comment:
So we could add a similar runtime object to the one `thread_pthread.h` uses. Might also be worth looking at `pthread_cond_clockwait` which allows passing the clock ID when the call is made rather than relying on the initializing the condition variable's attributes. It's was introduced with `sem_clockwait` in *glibc-2.30* and would work similarly to the semaphore implementation.

Reply:
> So we could add a similar runtime object to the one thread_pthread.h

I'd like to use the exact same fields as the ones used by `thread_pthread.h` and just refactor the initialization code to make it available to `Python/parking_lot.c`.

Something like:

In `pycore_pythread.h`:

```
// Get and maybe initialize _condattr_monotonic
static inline pthread_condattr_t _PyThread_get_condattr_monotonic(void) {...};
// Initializes _condattr_monotonic (implemented in `thread_pthread.h`)
extern pthread_condattr_t _PyThread_init_condattr_monotonic(void);
```

With the appropriate build guards (`_USE_PTHREADS` and `CONDATTR_MONOTONIC`) as needed.

> Might also be worth looking at pthread_cond_clockwait...

I'd like to stick with the APIs we're already using in `thread_pthread.h` to avoid any new platform specific issues. Also, when compiling with glibc, `parking_lot.c` only uses the semaphore code path (not pthreads.)

Reply:
On second thought, I don't think we should bother with `pthread_condattr_setclock` in `parking_lot.c`. The pthreads code path here is only used by macOS, and macOS doesn't support that call. Instead let's use [`pthread_cond_timedwait_relative_np`](https://github.com/apple-oss-distributions/libpthread/blob/d8c4e3c212553d3e0f5d76bb7d45a8acd61302dc/include/pthread/pthread.h#L556-L559) if available, which takes in a relative timeout and so avoids the clock issues. We'll need a `configure` check to check for it, but otherwise it should be more straightforward and effective.

Reply:
I'll get on that instead, explains why I couldn't figure out the point of `setclock` in testing.